### PR TITLE
enh: Adding contribution best practices in contribute docs

### LINKED ISSUE #16977: Start contributing page: Add section on how to create a good commit message and PR description
**This is a Feature Request**

**What would you like to be added**
Add section to Contributing docs on how to create a good commit message and PR description

**Why is this needed**
Good commit messages and PR descriptions are essential for 1) reviewers and approvers to understand the PR; 2) historical discovery when something goes wrong; and 3) when someone wants to see the history of a file. I see a lot of PR descriptions that are blank or just say "Fixes <issue-number>" instead of specifying actual changes to content or source code.

**Comments**
I wrote  [how to create a good commit message](https://wiki.acumos.org/display/AC/Your+First+Patch#YourFirstPatch-WritingYourCommitMessage) for another project. Adapt for K8s docs.

### PR DESCRIPTION
**Pull request by vishakhanihore**
Fixes #16977 

- This PR adds a `Contribution Best Practices` in Contribution Documents for PR Practices.

### CODE REVIEW

File: content/en/docs/contribute/_index.md [hunk 1]
```diff
@@ -71,6 +71,16 @@ documentation, but it should help you get started.
   - Propose improvements to docs tests
   - Propose improvements to the Kubernetes website or other tooling
 
+## Contributions best practices
+
+- Do write clear and meaningful GIT commit messages.
```
Comment (sftim) [inline content/en/docs/contribute/_index.md]: nit:
```suggestion
- Do write clear and meaningful Git commit messages.
```

File: content/en/docs/contribute/_index.md [hunk 2]
```diff
@@ -71,6 +71,16 @@ documentation, but it should help you get started.
...
+- Make sure to include _Github Special Keywords_ which references the issue and automatically closes the issue when PR is merged.
+- When you make a small change to a PR like fixing a typo, any style change, or changing grammar. Make sure you squash your commits so that you dont get a large number of commits for a relatively small change.
+- Make sure you include a nice PR description depicting the code you have changes, why to change a following piece of code and ensuring there is sufficient information for the reviewer to understand your PR.
+- Additional Readings : 
+    - [chris.beams.io/posts/git-commit/](https://chris.beams.io/posts/git-commit/)
+    - [github.com/blog/1506-closing-issues-via-pull-requests ](https://github.com/blog/1506-closing-issues-via-pull-requests )
+    - [davidwalsh.name/squash-commits-git ](https://davidwalsh.name/squash-commits-git )
```
Comment (sftim) [inline content/en/docs/contribute/_index.md]: @vishakhanihore how about adding this to https://kubernetes.io/docs/contribute/start/#improve-existing-content and / or https://kubernetes.io/docs/contribute/intermediate/ and then signposting readers there?

There is already a set of instructions on how to squash commits, for example.

(also, Prow auto-squashes commits for this repo if I'm remembering correctly)

Comment (vishakhanihore) [reply content/en/docs/contribute/_index.md]: @sftim Thanks for the suggestions, I will quickly look into it 👍

File: content/en/docs/contribute/_index.md [hunk 3]
```diff
@@ -71,6 +71,16 @@ documentation, but it should help you get started.
   - Propose improvements to docs tests
   - Propose improvements to the Kubernetes website or other tooling
 
+## Contributions best practices
+
+- Do write clear and meaningful GIT commit messages.
+- Make sure to include _Github Special Keywords_ which references the issue and automatically closes the issue when PR is merged.
```
Comment (sftim) [inline content/en/docs/contribute/_index.md]:
```suggestion
- You can include _GitHub Special Keywords_ which references the issue and automatically closes the issue when PR is merged.
```

**Pull request merged**